src/install/installer.ts

```diff
@@ -363,14 +363,55 @@ export class Installer {
         parentUrl
       );
 
-    // Find any existing locks in the current package scope - note that
-    // fallbacks to existing locks in other scopes are handled later:
+    // Fetch the current scope's pjson:
+    const definitelyPkgScope =
+      pkgScope || (await this.resolver.getPackageBase(parentUrl));
+    const pcfg =
+      (await this.resolver.getPackageConfig(definitelyPkgScope)) || {};
+
+    // By default, we take an install target from the current scope's pjson:
+    const pjsonTargetStr =
+      pcfg.dependencies?.[pkgName] ||
+      pcfg.peerDependencies?.[pkgName] ||
+      pcfg.optionalDependencies?.[pkgName] ||
+      (isTopLevel && pcfg.devDependencies?.[pkgName]);
+    const pjsonTarget =
+      pjsonTargetStr &&
+      newPackageTarget(
+        pjsonTargetStr,
+        new URL(definitelyPkgScope),
+        this.defaultRegistry,
+        pkgName
+      );
+    const resolutionInRange = async (resolution: InstalledResolution) => {
```

Comment:
This function checks that an existing resolution is in-range for the parent scope's `package.json` (if it has one).

Reply:
We are supposed to already have this function from the `inRange` function on the class itself - can we reuse / refactor into the existing implementation rather?

Reply:
Ah, right, I didn't see that - can do 👌